### PR TITLE
lib: at_custom_cmd and at_sms_cert: Add multiple AT commands and multiple filters support

### PR DESCRIPTION
**Pull request by eivindj-nordic**
At_filter library adds support for using AT filters in multiple
libraries simultaneously.

Add support to at_sms_cert library to support multiple commands on same call, required by new GCF baseline.

Fix init priority such that the library is initialized after the
nrf_modem_at_init on sys_init.

Libmodem: [PR-228](https://github.com/NordicPlayground/libmodem/pull/228)

### CODE REVIEW

File: lib/at_filter/at_filter.c [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+		return -NRF_E2BIG;
+	}
+	va_end(args);
+
+	return 0;
+}
+
+#if defined(CONFIG_AT_FILTER_SYS_INIT)
+/* Initialize during SYS_INIT */
+SYS_INIT(at_filter_sys_init, POST_KERNEL, 1);
```
Comment (lemrey) [inline lib/at_filter/at_filter.c:75]: should probably use: `APPLICATION` instead of `POST_KERNEL`.

File: lib/at_filter/at_filter.c [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+#include <nrf_modem_at.h>
+#include <modem/at_monitor.h>
+#include <toolchain/common.h>
+#include <stdbool.h>
+#include <nrf_errno.h>
+#include <modem/at_filter.h>
+#include <logging/log.h>
+#include <stdio.h>
+
+LOG_MODULE_REGISTER(at_monitor);
```
Comment (lemrey) [inline lib/at_filter/at_filter.c:19]: leftover

File: lib/at_filter/Kconfig [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+menuconfig AT_FILTER
+	bool "AT notification monitor"
```
Comment (lemrey) [inline lib/at_filter/Kconfig:8]: leftover

File: tests/lib/at_filter/src/main.c [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+	err = nrf_modem_at_printf("AT+CFUN=1");
+	zassert_equal(0, err, "nrf_modem_at_printf failed, error: %d", err);
+
+	/* Wait for network connection. */
+	do {
+		err = nrf_modem_at_cmd(response, sizeof(response), "AT+CEREG?");
+		zassert_equal(0, err, "nrf_modem_at_printf failed, error: %d", err);
+
+		err = sscanf(response, "\r\n+CEREG: %d,%d", &err, &connected);
+		zassert_equal(2, err, "sscanf failed, error: %d", err);
```
Comment (lemrey) [inline tests/lib/at_filter/src/main.c:64]: nit: you can use `nrf_modem_at_scanf` directly

**Review by lemrey — COMMENTED** [review]
Nice! I was wondering, since a "filter" is what we use to "redirect" an AT command to a custom implementation, perhaps we can find a better name for this library. In practice, what's happening is that the application is defining custom AT commands. What about: `at_custom_cmd`?

File: tests/lib/at_custom_cmd/prj.conf [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+CONFIG_ZTEST=y
+
+# NewLib C
+CONFIG_NEWLIB_LIBC=y
+
+# Enable and initialize modem library
+CONFIG_NRF_MODEM_LIB=y
+CONFIG_NRF_MODEM_LIB_SYS_INIT=y
+
+#Enable AT SMS certification support
```
Comment (VeijoPesonen) [inline tests/lib/at_custom_cmd/prj.conf:17]: Comment implies that CONFIG_AT_SMS_CERT would be enabled instead of CONFIG_AT_CUSTOM_CMD only.

Comment (eivindj-nordic) [reply tests/lib/at_custom_cmd/prj.conf:17]: Yes, this should be "Enable custom AT commands". AT SMS certification support has its own tests.

File: lib/at_sms_cert/at_sms_cert.c [hunk 6]
```diff
@@ -0,0 +1,400 @@
...
+	}
+
+	return NULL;
+}
+
+/* Internal filter with multiple command support */
+
+int at_sms_cert_filter_callback(char *buf, size_t len, char *at_cmd)
+{
+
```
Comment (anhmolt) [inline lib/at_sms_cert/at_sms_cert.c:114]: More whitespace

**Pull request closed without merging**